### ISSUE TITLE
Missing yaml tags/serialization bug?

### ISSUE THREAD

**Issue by pmichel31415** [neulab/xnmt#368]
I couldn't load a model containing a ResidualBiRNNBuilder, and I realized that the class didn't have a `yaml_tag` field.

There should be tests in place to prevent this from happening.

So here's the error message:
```python
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!ResidualBiRNNBuilder'
  in "recipes/models/europarl-en-fr.mod", line 106, column 14
```

**Comment by pmichel31415** [comment]
Hmm this is very unfortunate, I'll tag it as major bug

**Comment by neubig** [comment]
Is it fixed by https://github.com/neulab/xnmt/pull/371 ?

(really sorry about this being broken)

**Comment by pmichel31415** [comment]
The BiRNNBuilder part is fixed by #371 

The part about LinearBridge not being serializable isn't. Also IMO there should be automated tests so that this last bug doesn't happen in the future with other modules

**Comment by msperber** [comment]
This is fixed by #372 so I'll close.
